packages/vite-plugin-checker/src/checkers/typescript/main.ts

```diff
@@ -123,12 +123,21 @@ export class TscChecker extends Checker<'typescript'> {
       absFilePath: __filename,
       build: {
         buildBin: (config) => {
-          if (typeof config.typescript === 'object' && config.typescript.tsconfigPath) {
-            const tsconfig = config.typescript.root
-              ? path.join(config.typescript.root, config.typescript.tsconfigPath)
-              : config.typescript.tsconfigPath
-            return ['tsc', ['--noEmit', '-p', tsconfig]]
+          if (typeof config.typescript === 'object') {
+            const { root, tsconfigPath, buildMode } = config.typescript
+
+            // Compiler option '--noEmit' may not be used with '--build'
+            let args = [buildMode ? '-b' : '--noEmit']
```

Comment:
If noEmit is not supported in build mode. The checker will bring side effects(compiled files), I'm a little concerned about it. What do you think 🤔

Reply:
I'm looking for this functionality and I see no problem with outputting files. In fact, this is the way it has to work because this is the way `tsc` works in incremental mode and you cannot run `--build` without incremental mode. No need to clean. Just make a note in the docs that files will be output and it should be fine IMO.

Reply:
Reference: https://github.com/microsoft/TypeScript/issues/36917 (composite mode is what --build uses).

Reply:
@fi3ework Is there a chance it will be merged?